wordpress-mobile/WordPress-Android
# Issue/13326 ignore threat

Parent #13326 

This PR handles ignore threat action on the threat details screen.
Child PR for `IgnoreThreatUseCase`: #13767

It mimics behaviour on the web:
1. Displays a confirmation dialog 
2. Disables buttons while action is taking place
3. Displays success/ error messages when action completes
4. Refreshes scan state when ignore threat action is successful

https://user-images.githubusercontent.com/1405144/104564678-37a84580-5671-11eb-8893-266e40bf6554.mov

https://user-images.githubusercontent.com/1405144/104564816-60c8d600-5671-11eb-8c6a-d4b9084eb60e.mov

To test:
Prerequisite: 

- Make sure both Scan feature flag and MySiteImprovements flag are set to on in the App settings.
- You have access to WP.com account with a site having scan capability and threats (e.g. https://pressable-jetpack-daily-scan.mystagingwebsite.com). Server credentials should be set on the site.
- I've added a threat to the site. In case you run out of threats, re-install bad calendar plugin from [here](https://jetpackp2.files.wordpress.com/2020/05/calendar-1.zip) and re-run scan.

1. Login to the app using above WP.com account and select the site on the My Site tab
2. Click scan menu item on My Site
3. Notice that scan state is fetched and shown with threats
4. Click a threat 
5. Click ignore threat button on the threat details screen
6. Notice that a _confirmation dialog is shown_ (see video)
7. Click ok in the confirmation dialog
8. Notice that _action buttons are disabled_
8. Wait for some time till ignore threat action completes
9. Notice that _success message "Threat ignored." is shown_  
10. Notice that _update to scan state is triggered after ~2 secs

Repeat above steps and enable airplane mode after step 6. Notice that an error message is displayed and actions buttons are re-enabled.

(Tests included in `ThreatDetailsViewModelTest` )

**Notes**
1. Ignore all styling
2. Server credentials check is not handled
3. Loading state while action is taking place - yet to be added

PR submission checklist:
